### ISSUE TITLE
A new linter based on match.call

### ISSUE THREAD

**Issue by renkun-ken** [jimhester/lintr#837]
I don't see an existing linter checks if the arguments of the call are valid (correct argument names or unused argument)

```r
> e <- quote(assign("x", 1 + 1, enivr = globalenv()))                                                                                              

> call <- match.call(assign, e)                                                                                                                    
Error in match.call(definition, call, expand.dots, envir) : 
  unused argument (enivr = globalenv())
Backtrace:
1: match.call(assign, e)
```

**Comment by AshesITR** [comment]
When implementing this we should keep in mind that functions might be redefined and thus there is potential for false positives as well as false negatives.

**Comment by MichaelChirico** [comment]
Do we have a fixed idea of what sorts of linters we want in `lintr`, vs those that belong in extension packages?

My thinking so far has been to try and stick to things explicitly laid out in the tidyverse style guide (https://style.tidyverse.org/)

**Comment by renkun-ken** [comment]
I think the purpose of the package in general should include

* Conform to a popular style guide
* Discover potential problems early

**Comment by AshesITR** [comment]
I think if we deem a linter could be useful, we should add it.
Regarding the default configuration, that should stick to the tidyverse styleguide for style matters and not include linters with potential false positives IMO.
